docs/checks/promql/rate.md

```diff
@@ -30,6 +30,10 @@ to verify that:
   See [this blog post](https://www.robustperception.io/rate-then-sum-never-sum-then-rate/)
   for details.
 
+The check uses the `/api/v1/status/config` Prometheus endpoint, if you are
```

Comment:
> This is not really specific to `promql/rate`, there are more checks that interact with various Prometheus APIs and will fail if the target service doesn't implement them.

Ah! Thank you for the clarification, that makes sense

> pint could potentially check if each defined `prometheus` instance supports all API endpoint and if not A) warn B) disable checks that use it

Yeah definitely at least a warning is in order. I'm also wondering if for cases like Thanos we could synthesize the required configuration in pint? e.g. a `prometheus` section can also ship with additional configuration to be used in lieu of the API endpoint

Reply:
Yes, being able to compensate for lack of APIs via extra config sounds like a good idea.

That being said current promql/rate logic is incomplete, it only reads global scrape_interval and doesn't check each scrape job scrape_interval, so it might produce false results in some cases. I just never had time to finish that.

What it means is that extra config could lead to same issues with invalid results, which might be a reason to add per rule/file scrape_interval overrides.
My only hesitation here would be that pint is suppose to check things for you, if it only works if you explicitly submit a lot of information then that misses the point. Scrape interval could change in Prometheus config while staying the same in pint config potentially causing issues that pint wouldn't notice.

Reply:
Could we convert this to an issue please?

Reply:
Raised #575 with some extra docs as an alternative to this PR